### PR TITLE
add checks for newest and minimum required pip version

### PR DESCRIPTION
**Pull request by MalteEbner**
close: #204

### CODE REVIEW

File: lightly/api/version_checking.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+def get_minimum_compatible_version():
+    versioning_api = get_versioning_api()
+    version_number: str = versioning_api.get_minimum_compatible_pip_version()
+    return version_number
+
+
+def version_compare(v0, v1):
+    v0 = [int(n) for n in v0.split('.')][::-1]
+    v1 = [int(n) for n in v1.split('.')][::-1]
+    assert len(v0) == 3
```
Comment (philippmwirth) [inline lightly/api/version_checking.py:40]: we shouldn't use asserts in production code

File: lightly/api/version_checking.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+def get_minimum_compatible_version():
+    versioning_api = get_versioning_api()
+    version_number: str = versioning_api.get_minimum_compatible_pip_version()
+    return version_number
+
+
+def version_compare(v0, v1):
+    v0 = [int(n) for n in v0.split('.')][::-1]
+    v1 = [int(n) for n in v1.split('.')][::-1]
```
Comment (philippmwirth) [inline lightly/api/version_checking.py:39]: This works for the moment but maybe we'll have to find a better way to do this...

File: lightly/api/api_workflow_client.py [hunk 3]
```diff
@@ -67,6 +72,13 @@ def __init__(self, token: str, dataset_id: str = None, embedding_id: str = None)
         self.scores_api = ScoresApi(api_client=api_client)
         self.samples_api = SamplesApi(api_client=api_client)
 
+    def check_version_compatibility(self):
+        minimum_version = get_minimum_compatible_version()
+        if version_compare(__version__, minimum_version) < 0:
+            raise ValueError(f"Incompatible Version of lightly pip package. "
+                             f"Please upgrade to at least version{minimum_version} "
```
Comment (philippmwirth) [inline lightly/api/api_workflow_client.py:79]: Add a space between `version` and `{minimum_version}` or do it like this `v{minimum_version}`

File: lightly/api/api_workflow_client.py [hunk 4]
```diff
@@ -67,6 +72,13 @@ def __init__(self, token: str, dataset_id: str = None, embedding_id: str = None)
         self.scores_api = ScoresApi(api_client=api_client)
         self.samples_api = SamplesApi(api_client=api_client)
 
+    def check_version_compatibility(self):
+        minimum_version = get_minimum_compatible_version()
+        if version_compare(__version__, minimum_version) < 0:
```
Comment (philippmwirth) [inline lightly/api/api_workflow_client.py:77]: We also need some environment variable here (e.g. `LIGHTLY_FORCE_CONNECTION`) as described in the issue. This should disable the exception.

File: lightly/api/version_checking.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        if x < y:
+            return -1
+        if x > y:
+            return 1
+    return 0
+
+
+def pretty_print_latest_version(latest_version, width=70):
+    lines = [
+        'There is a newer version of the package available.',
```
Comment (philippmwirth) [inline lightly/api/version_checking.py:53]: Maybe we can replace this by a warning. Low-prio though

File: lightly/api/version_checking.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+from lightly.openapi_generated.swagger_client.api_client import ApiClient
+
+from lightly.openapi_generated.swagger_client.configuration import Configuration
+from lightly.api.utils import getenv
+
+
+def get_versioning_api() -> VersioningApi:
+    configuration = Configuration()
+    configuration.host = getenv('LIGHTLY_SERVER_LOCATION', 'https://api.lightly.ai')
+    token = getenv('TOKEN', None)
```
Comment (philippmwirth) [inline lightly/api/version_checking.py:15]: no need for a token, the routes are whitelisted.

File: lightly/api/__init__.py [hunk 7]
```diff
@@ -1,8 +0,0 @@
-""" The lightly.api module provides access to the Lightly web-app. """
```
Comment (philippmwirth) [inline lightly/api/__init__.py:1]: Please add this file again, even if it's empty we need it in the docs.

File: lightly/api/api_workflow_client.py (hunk 4 above)
Comment (IgorSusmelj) [reply lightly/api/api_workflow_client.py:77]: let's make another issue out of it and push it with the next update.

**Review by philippmwirth — APPROVED** [review]

**Pull request merged**